### DISCUSSION TITLE
What JSON library to use?

### DISCUSSION THREAD

**Discussion by JxBP** [hypixelapiwrapper/hypixel-api-java#38] [Q&A] [answered]
So as people mentioned we should choose wisely so we dont have to go back later and refactor everything to adopt a new library and since the api's responses can be quite large parsing could drastically slow things down.

Candidates currently are:
- org.json
- gson
- Jackson (probably the fastest)

Another idea would be to let the user choose once again similar to how our caching strategies work.

**Comment by Netwyn** [comment]
`Jackson` is most likely the best; Letting the user choose would be extremely complicated
`org.json` was originally used for simplicity and because `Jackson` is confusing as hell, but it was always what I had "wanted to do".

**Reply by unpluggedsam** [reply]
I like it. We should get started as soon as possible so we can refactor minimal code.

**Reply by TheNullicorn** [reply]
> Letting the user choose would be extremely complicated

I don't believe it would be? We would just create an abstraction over JSON objects/arrays (e.g. `HypixelObject`) and the parser, then provide default implementations for libraries in separate gradle subprojects

**Comment by unpluggedsam** [comment, marked as answer]
We went with org.json
